### PR TITLE
Viite 1591 unable to reserve road part

### PR DESCRIPTION
**Pull request by Campanha**
Modify Project_link column "Geometry" from VARCHAR to SDO_GEOMETRY type and changed all CRUD operations due to that.

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/oracle/OracleDatabase.scala [hunk 1]
```diff
@@ -107,4 +110,23 @@ object OracleDatabase {
                          ) = 'TRUE'
     """
   }
+
+  def create(points: Seq[Point], con: java.sql.Connection): STRUCT = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/oracle/OracleDatabase.scala]: Should change the name of it to `createJGeometry`.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/oracle/OracleDatabase.scala [hunk 2]
```diff
@@ -107,4 +110,23 @@ object OracleDatabase {
...
+
+  def create(points: Seq[Point], con: java.sql.Connection): STRUCT = {
+    val ordinates = points.flatMap(p => Seq(p.x, p.y, p.z)).toArray
+    val dim = 4
+    val srid = 3067
+    val oracleConn = dynamicSession.conn.asInstanceOf[ConnectionHandle].getInternalConnection
+      JGeometry.store(JGeometry.createLinearLineString(ordinates, dim, srid), oracleConn)
+  }
+
+  def load(geometry: Option[Object]): Seq[Point] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/oracle/OracleDatabase.scala]: Should change the name to something like `convertJGeometryToGeometry`.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala [hunk 3]
```diff
@@ -284,76 +285,85 @@ object ProjectDAO {
...
-    else
-      Seq()
-  }
-
   private def listQuery(query: String) = {
     Q.queryNA[ProjectLink](query).iterator.toSeq
   }
 
+  //TODO delete when there is no longer needed. This is now need to script where we migrate PROJECT_LINK .GEOMETRY_STRING to .GEOMETRY
+  def updateGeometryStringToSdo(id: Long, geometry: Seq[Point]): Unit = {
```
Comment (leirasf) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala]: Now I think is no longer needed 🥇

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala [hunk 4]
```diff
@@ -169,6 +178,10 @@ package object viite {
     }.toSeq
   }
 
+  def toGeomDouble(point: Point): Seq[Double] = {
```
Comment (leirasf) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala]: This method can belong to the Point case class.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala [hunk 5]
```diff
@@ -143,6 +145,13 @@ package object viite {
     isRamp(r.roadNumber, r.track.value)
   }
 
+  def parseStringGeometry(geomString: String): Seq[Point] = {
```
Comment (leirasf) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/package.scala]: This shouldn't be needed anymore.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala (hunk 3 above)
Comment (Campanha) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectDAO.scala]: This one will be used to the batch to migrate PROJETC_LINK(_HISTORY) Geometry (VARCHAR (4000 BYTE)) to Geometry (SDO_GEOMETRY). After that the batch and its code will be deprecated.

**Pull request merged**